ksqldb-engine/src/test/java/io/confluent/ksql/query/QueryBuilderTest.java

```diff
@@ -321,6 +323,11 @@ public void shouldBuildCreateAsPersistentQueryCorrectly() {
     assertThat(queryMetadata.getProcessingLogger(), equalTo(uncaughtProcessingLogger));
     assertThat(queryMetadata.getPersistentQueryType(),
         equalTo(KsqlConstants.PersistentQueryType.CREATE_AS));
+    // queries in dedicated runtimes must not include alternative topic prefix
+    assertThat(
+        queryMetadata.getStreamsProperties().get(InternalConfig.TOPIC_PREFIX_ALTERNATIVE),
+        is(nullValue())
+    );
```

Comment:
I'd suggest that we add a unit test to show precisely what we expect the topic names to be.  

Maybe that's a bit prescriptive/specific.

Reply:
No we should absolutely do this 

also do we have unit tests for all cases/types of queries? Given how many bugs like this have been found so far due to the limited coverage, particularly end-to-end, of ksql unit/integration tests...well, we should aim to fill in all the holes for a given fix/PR